Test_oM/UnitTests/UnitTest.cs

```diff
@@ -0,0 +1,46 @@
+﻿/*	
+ * This file is part of the Buildings and Habitats object Model (BHoM)	
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.	
+ *	
+ * Each contributor holds copyright over their respective contributions.	
+ * The project versioning (Git) records all such contribution source information.	
+ *                                           	
+ *                                                                              	
+ * The BHoM is free software: you can redistribute it and/or modify         	
+ * it under the terms of the GNU Lesser General Public License as published by  	
+ * the Free Software Foundation, either version 3.0 of the License, or          	
+ * (at your option) any later version.                                          	
+ *                                                                              	
+ * The BHoM is distributed in the hope that it will be useful,              	
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               	
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 	
+ * GNU Lesser General Public License for more details.                          	
+ *                                                                            	
+ * You should have received a copy of the GNU Lesser General Public License     	
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      	
+ */
+
+using BH.oM.Base;
+using System.Collections.Generic;
+using System.Reflection;
+using System.ComponentModel;
+
+namespace BH.oM.Test.UnitTests
+{
+    [Description("Defines a datadriven unit test for a specific method. Class defines inputs, method to be run and expected output.")]
+    public class UnitTest : BHoMObject
+    {
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+
+        [Description("The method to be tested.")]
```

Comment:
Might be helpful to give some additional information - like how should the `MethodBase` be provided? Or is it expecting only BHoM methods? Etc.? (Feel free to disregard this one, this is just a suggestion and I know I'm not the best at discussing descriptions :wink: )